weatherman.py

```diff
@@ -0,0 +1,208 @@
+            self.report_printer.print_average_temperatures_month(average_values)
+
+    def generate_barchart_for_file_controller(self, file_name):
+        weather_data = self.weather_files_handler.return_weather_data(file_name)
+        if weather_data:
+            self.report_printer.print_temperatures_bar_for_each_record(weather_data)
+    
+    def track_tempratures(self,weather_data):
+        for line_read in weather_data: 
+            self.temperature_tracker.track_min_temperature(line_read["Min TemperatureC"], line_read["PKT"])
+            self.temperature_tracker.track_max_temperature(line_read["Max TemperatureC"], line_read["PKT"])
+            self.temperature_tracker.track_max_humidity(line_read["Max Humidity"], line_read["PKT"])
+
+    def generate_report_for_year_controller(self, year):
+        files_name_list = self.weather_files_handler.return_all_file_names_list_for_year(year)
+        for file_name in files_name_list:
+             weather_data = self.weather_files_handler.return_weather_data(file_name)
+             if weather_data:
+                self.track_tempratures(weather_data)                
+        self.temperature_tracker.print_temperatures_year()
+
+
+class CmdArgumentsHandler():
+
+    report_generator = ReportGenerator()
+
+    def validate_year_month(self, year, month_num): 
+        return year.isnumeric() and month_num.isnumeric() and int(month_num) in range (1, 13)
+    
+    def handle_cmd_argument_year(self, year):
+        self.report_generator.generate_report_for_year_controller(year)
+
+    def get_year_and_month_from_arguments(self, cmd_argument):
+        input = cmd_argument.split('/')                
+        input_dict = {"year": input[0], 
+                      "month": input[1]}
+        return input_dict
+    
+    def handle_cmd_arguments(self, cmd_argument, key):    
+        input_dict = self.get_year_and_month_from_arguments(cmd_argument)
+        if self.validate_year_month(input_dict["year"], input_dict["month"]): 
+
+            input_dict["month"] = calendar.month_abbr[int(input_dict["month"])]       
+            file_name = WeatherFilesHandler.match_and_return_file_name(input_dict["year"], input_dict["month"])    
+
+            if file_name and key == 'a' :
+                self.report_generator.generate_report_for_month_controller(file_name)
+
+            elif file_name and key == 'c' :
+                self.report_generator.generate_barchart_for_file_controller(file_name)              
+        else: 
+            print("INVALID Year or Month Entered")
+
+
+def main():
+
+    cmd_handler = CmdArgumentsHandler()
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-e', help="To print Report for year")
+    parser.add_argument('-a', help="To print AVG Report ")
+    parser.add_argument('-c', help="To print Barchart Report")
+    args = parser.parse_args()
+
+    input_e = args.e
+    input_a = args.a
+    input_c = args.c
```

Comment:
Why are we not using them directly below? What is the point of having them in an extra variable?